plugins/eni/engine/dhclient.go

```diff
@@ -131,14 +144,29 @@ func (client *dhclient) IsExecutableInPath() bool {
 
 func (client *dhclient) Start(deviceName string, ipRev int) error {
 	args := client.constructDHClientArgs(deviceName, ipRev)
-	cmd := client.exec.Command(client.executable, args...)
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		log.Errorf("Error executing '%s' with args '%v': raw output: %s",
-			client.executable, args, string(out))
-		return errors.Wrapf(err,
-			"engine dhclient: unable to start dhclient for ipv%d address; command: %s %v; output: %s",
-			ipRev, client.executable, args, string(out))
+	var out []byte
+	var err error
+	attempts := 1
+	for {
```

Comment:
You're doing the same kind of retry in two different ways.  Maybe make them the same?  In particular:

* You can use the attempt count in the for loop itself `for attempts := 0; attempts < client.dhclientStartAttepmtsMax; attempts++`
* You can check for `err != nil` outside the loop and use that to log or return an error from the function